src/compiler/commandLineParser.ts

```diff
@@ -231,6 +231,13 @@ namespace ts {
             category: Diagnostics.Basic_Options,
             description: Diagnostics.Do_not_emit_outputs,
         },
+        {
+            name: "noEmitJs",
+            type: "boolean",
+            showInSimplifiedHelpView: true,
+            category: Diagnostics.Basic_Options,

```

Comment:
i would say this belongs into `Advanced_Options`

Reply:
AdvancedOptions won't show up in `--help` and this is something that will definitely be very useful with allowJs and checkJs projects to generate their typings. Basically enabling thousands of npm js repos to automatically generate typings if they've annotated code with jsdoc.

Although the PR doesn't include full understanding of prototype based functions. Does it work with you if we put this in experimental mode first ?

Reply:
Advanced_options don't show up as part of `--help` or in default tsc --init. Doesn't it make sense that the option showed up there? I plan to enable allowJs + emitDeclarationsOnly in another PR which would help with lots of npm projects to auto-generate jsdoc to .d.ts declarations